logging/impl_test.go

```diff
@@ -190,3 +191,29 @@ func TestConsoleOutputFormat(t *testing.T) {
 	assertLogMatches(t, notStdout,
 		`2023-10-31T14:25:49.124Z	INFO	impl	logging/impl_test.go:177	impl logw	{"key":"val","fmt.Sprintf":"{x:1 y:{Y1:y1} Z:z}"}`)
 }
+
+func TestContextLogging(t *testing.T) {
+	ctx := context.Background()
+
+	// A logger object that will write to the `notStdout` buffer.
+	const inUTC = true
+	notStdout := &bytes.Buffer{}
+	logger := &impl{"impl", NewAtomicLevelAt(INFO), inUTC, []Appender{NewWriterAppender(notStdout)}}
+
+	logger.CDebug(ctx, "Debug log")
+	test.That(t, notStdout.Len(), test.ShouldEqual, 0)
+
+	traceKey := "foobar"
+	ctx = EnableDebugModeWithKey(ctx, traceKey)
+	logger.CDebug(ctx, "Debug log")
+	assertLogMatches(t, notStdout,
+		`2023-10-30T09:12:09.459Z	DEBUG	impl	logging/impl_test.go:200	Debug log	{"traceKey":"foobar"}`)
+
+	logger.CDebugf(ctx, "Debugf log %v", "debugf")
+	assertLogMatches(t, notStdout,
+		`2023-10-30T09:12:09.459Z	DEBUG	impl	logging/impl_test.go:200	Debugf log debugf	{"traceKey":"foobar"}`)
+
+	logger.CDebugw(ctx, "Debugw log", "key", "value")
```

Comment:
[nit] Could also test info, warn and error levels.

Reply:
We could :P

Done